File: proselint/checks/misc/back_formations.py
"""
Back-formations.

---
layout:     post
source:     Garner's Modern American Usage
source_url: http://bit.ly/1T4alrY
title:      back-formations
date:       2014-06-10 12:31:19
categories: writing
---

Back-formations.

"""

from proselint.registry.checks import Check, types

check = Check(
    check_type=types.PreferredFormsSimple(items={"improprietous": "improper"}),
    path="misc.back_formations",
    message="Back-formation. '{}' is the preferred form.",
)

__register__ = (check,)

File: proselint-rs/crates/proselint-checks/src/misc/back_formations.rs
use proselint_registry::checks::{Check, types::*, Padding};
use phf::phf_map;

pub const EXAMPLES_PASS: &[&str] = &["Smoke phrase with nothing flagged."];
pub const EXAMPLES_FAIL: &[&str] = &["It is an improprietous use."];

const CHECK: Check = Check {
	check_type: &PreferredFormsSimple {
		items: &phf_map!("improprietous" => "improper"),
		padding: Padding::WordsInText,
	},
	path: "misc.back_formations",
	msg: "Back-formation. '{}' is the preferred form.",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];
